extensions/objects/templates/literally-canvas-diagram-editor.component.spec.ts

```diff
@@ -59,9 +134,264 @@ describe('LiterallyCanvasDiagramEditor', function() {
     expect(LCDiagramEditorCtrl.getDiagramSizeInfo()).toBe(helpText);
   });
 
-  it('should validate data', function() {
-    // TODO(#9357): Will be implemented once the svg data saving
-    // functionality is implemented.
-    expect(LCDiagramEditorCtrl.validate()).toBe(false);
+  it('should check if diagram is created', function() {
+    LCDiagramEditorCtrl.lc.currentSvg = defaultsvg;
+    expect(LCDiagramEditorCtrl.isDiagramCreated()).toBe(false);
+    LCDiagramEditorCtrl.lc.currentSvg = linesvg;
+    expect(LCDiagramEditorCtrl.isDiagramCreated()).toBe(true);
+  });
+
+  it('should check whether user is drawing', function() {
+    expect(LCDiagramEditorCtrl.isUserDrawing()).toBe(false);
+    LCDiagramEditorCtrl.lc._shapesInProgress = ['tool'];
```

Comment:
Should not call private methods in tests. Can you find another way to set this?

Reply:
Done.